api/opencl_architecture.asciidoc

```diff
@@ -2104,6 +2104,20 @@ The new version of OpenCL C must be explicitly requested via the `-cl-std=` buil
 Whenever an OpenCL C feature is optional in the new version of the OpenCL C programming language, it will be paired with a feature macro, such as `+__opencl_c_feature_name+`, and a corresponding API query.
 If a feature macro is defined then the feature is supported by the OpenCL C compiler, otherwise the optional feature is not supported.
 
+In order to allow future versions of OpenCL to support new types of
+devices, some capabilities that are currently required of all OpenCL
+devices may become optional in future versions of OpenCL.
+When a required capability becomes optional, all existing profiles
+will continue to require that it be supported, thereby guaranteeing
+that applications targeting any of these profiles will continue to run
+with subsequent minor releases of OpenCL.
+New profiles that are introduced in or after the version of the
+standard that made the capability optional will not be required to
+support the capability.
```

Comment:
This is fairly nitpicky, but:

To me, the key takeaways from this paragraph should be:

* Within a profile, no features or capabilities that are currently required will become optional, except perhaps in a major API revision (and even then, this will hopefully be rare).
* New profiles may be added in minor API revisions with different required features or capabilities to enable new classes of devices.

Right now, the first sentence doesn't convey either of these key takeaways, and the next two sentences are necessary to clarify when a required capability may become optional.  I'd consider swapping the first two sentences so there's less room for possible confusion.  How about something like:

> All features that are required for an OpenCL profile will also be required in subsequent minor releases of OpenCL, thereby guaranteeing backwards compatibility for applications targeting specific profiles.  However, in order to allow future versions of OpenCL to support new devices, minor releases of OpenCL may add new profiles where some features that are currently required for all OpenCL profiles becomes optional.  It is therefore strongly recommended that applications query the profile supported by the OpenCL device they are running on in order to remain robust to future changes.

This is just a suggestion and these changes can go in as-is if needed.